### PR TITLE
Enable cuFile on ARM

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/rapidsai/kvikio/issues/339

Builds CUDA 12.2+ ARM packages with cuFile and adds cuFile as a dependency. Continues to build CUDA 11 ARM packages without cuFile.

As cuFile already has the needed logic to `dlopen` cuFile at runtime (with fallbacks for when it is missing), there shouldn't be any additional handling needed for legacy CUDA versions with ARM support.

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -207,12 +207,18 @@ dependencies:
         matrices:
           - matrix:
               arch: aarch64
+              cuda: "12.*"
```
[the remark this answers is not in the document]
Comment (jakirkham) [reply dependencies.yaml]: Agreed. Had tried this before, but ran into some issues. Though this may have been caused by other changes. Appears to work now

**Review by madsbk — APPROVED** [review]
Nice, thanks @jakirkham !

File: dependencies.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ files:
     output: conda
     matrix:
       cuda: ["11.8", "12.5"]
-      arch: [x86_64]
+      arch: [aarch64, x86_64]
```
Comment (bdice) [inline dependencies.yaml:7]: I would prefer to revert this. We don't supply ARM environment files for most RAPIDS repos. If you'd like to add ARM conda envs for all of RAPIDS, I would do that in a separate PR and track the effort in rapidsai/build-planning for all repos.
```suggestion
      arch: [x86_64]
```

Comment (jakirkham) [reply dependencies.yaml:7]: Given there are other users asking about ARM specifically with KvikIO (and running into issues without more explicit guidance), the environments provide a value add for them. For example: https://github.com/rapidsai/kvikio/issues/393

File: dependencies.yaml [hunk 3]
```diff
@@ -207,12 +207,18 @@ dependencies:
...
+              cuda: "12.*"
             packages:
+              - libcufile-dev
           - matrix:
-              cuda: "12.*"
               arch: x86_64
+              cuda: "12.*"
             packages:
               - libcufile-dev
+          - matrix:
```
[the remark this answers is not in the document]
Comment (msarahan) [reply dependencies.yaml:216]: I see, I didn't realize that we had to be so explicit. Thanks for clarifying that.

**Review by msarahan — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

Anything else needed here?

File: dependencies.yaml (hunk 2 above)
Comment (bdice) [reply dependencies.yaml:7]: I suppose it's fine to include this. Maybe this should be done for all of RAPIDS, with a build-planning issue for tracking.

We don't have a good way to exclude (CUDA 11 + ARM), unless you create a separate file key entry for (CUDA 12 + ARM) That's not worth doing imo.

As we see increased ARM usage, it'll be somewhat helpful to have these environment files readily accessible. I have had to explain how to generate the ARM environments with `rapids-dependency-file-generator` before and it's a little complicated for new DFG users.

**Review by bdice — APPROVED** [review]

File: dependencies.yaml (hunk 2 above)
Comment (jakirkham) [reply dependencies.yaml:7]: Agree this is a good idea

Have raised issue: https://github.com/rapidsai/build-planning/issues/87

Please add anything additional to that
